relay-server/src/actors/upstream.rs

```diff
@@ -246,21 +325,71 @@ impl UpstreamRelay {
             //   1. Have own connection pool for `store` requests
             //   2. Buffer up/queue/synchronize events before creating the request
             .wait_timeout(self.config.event_buffer_expiry())
-            .conn_timeout(self.config.http_connection_timeout())
             // This is the timeout after wait + connect.
             .timeout(self.config.http_timeout())
+            .conn_timeout(self.config.http_connection_timeout())
+            .to_tracked(self.http_request_finished_notifier.clone())
             .map_err(UpstreamRequestError::SendFailed)
-            .and_then(handle_response);
+            .and_then(handle_response)
+            .then(|x| {
+                response_sender.send(x).ok();
+                futures::future::ok(())
```

Comment:
_nit_: You can simplify this by returning `Ok(())`, which will be automatically transformed into a future.

Reply:
ok